### DISCUSSION TITLE
How to use TTS from the Python library instead of from the terminal?

### DISCUSSION THREAD

**Discussion by SaadBazaz** [coqui-ai/TTS#1596] [General Q&A]
Hi there,
I'm having a tough time finding this in the docs. It should be in the Synthezining Speech documentation, afaik.

How can I synthesize speech directly from the library? i.e.
```
import TTS

TTS.synthesize( #args )
```

Currently I call the CLI commands in Python using subprocess.

**Comment by thorstenMueller** [comment]
You can run the TTS-server process and use the cTTS Python module to synthesize voice within any Python application.
https://pypi.org/project/cTTS/

Maybe this helps.

**Reply by SaadBazaz** [reply]
That's a nice workaround, but still a bit junky imo. Can't add other arguments like reference speaker WAV...

Haven't the official Coqui TTS done anything about this? Maybe this should be a feature request?

**Comment by lexkoro** [comment]
You can use this https://github.com/coqui-ai/TTS/blob/dev/TTS/bin/synthesize.py as reference.

**Reply by SaadBazaz** [reply]
This looks like the right way to go. Thanks!
